torch/onnx/utils.py

```diff
@@ -1844,6 +1844,30 @@ def _run_symbolic_function(
         env=env,
     )
 
+    # Direct ATen export requested
+    # For BUILD_CAFFE2=0 builds, if domain=="aten" and operator_export_type==ONNX_ATEN,
+    #   an aten::ATen operator is created regardless of symbolics existence
+    # For BUILD_CAFFE2=1, the same applies only if there is no symbolic available
```

Comment:
Related to the CI failure on `test_py_diagnose_emits_error`, where it expects a `operator_supported_in_newer_opset_version` error for `aten::diagonal` when exporting in opset 9. After this change, the error was not thrown, and `aten::diagonal` is exported as `aten::ATen(...)`. The `operator_export_type` of `export` is the default `_C_onnx.OperatorExportTypes.ONNX`.

Is this behavior expected, for BUILD_CAFFE2=1?

Reply:
running 

```python
def test_py_diagnose_emits_error():
    class M(torch.nn.Module):
        def forward(self, x):
            return torch.diagonal(x)

    torch.onnx.export(
        M(),
        torch.randn(3, 4),
        io.BytesIO(),
        opset_version=9,
        verbose=True
    )

test_py_diagnose_emits_error()
```

on a build with `BUILD_CAFFE2=1` locally I got

```bash
Traceback (most recent call last):
  File "/pytorch_repros/bad_aten_mode.py", line 57, in <module>
    test_py_diagnose_emits_error()
  File "/pytorch_repros/bad_aten_mode.py", line 49, in test_py_diagnose_emits_error
    torch.onnx.export(
  File "/github/pytorch-dev2/torch/onnx/__init__.py", line 305, in export
    return utils.export(model, args, f, export_params, verbose, training,
  File "/github/pytorch-dev2/torch/onnx/utils.py", line 118, in export
    _export(model, args, f, export_params, verbose, training, input_names, output_names,
  File "/github/pytorch-dev2/torch/onnx/utils.py", line 719, in _export
    _model_to_graph(model, args, verbose, input_names,
  File "/github/pytorch-dev2/torch/onnx/utils.py", line 503, in _model_to_graph
    graph = _optimize_graph(graph, operator_export_type,
  File "/github/pytorch-dev2/torch/onnx/utils.py", line 232, in _optimize_graph
    graph = torch._C._jit_pass_onnx(graph, operator_export_type)
  File "/github/pytorch-dev2/torch/onnx/__init__.py", line 354, in _run_symbolic_function
    return utils._run_symbolic_function(*args, **kwargs)
  File "/github/pytorch-dev2/torch/onnx/utils.py", line 1057, in _run_symbolic_function
    symbolic_fn = _find_symbolic_in_registry(domain, op_name, opset_version, operator_export_type)
  File "/github/pytorch-dev2/torch/onnx/utils.py", line 1011, in _find_symbolic_in_registry
    return sym_registry.get_registered_op(op_name, domain, opset_version)
  File "/github/pytorch-dev2/torch/onnx/symbolic_registry.py", line 129, in get_registered_op
    raise RuntimeError(msg)
RuntimeError: Exporting the operator diagonal to ONNX opset version 9 is not supported. Support for this operator was added in version 13, try exporting with this version.
```

Isn't that the expected behavior?

Reply:
The difference from the original behavior (after aten fallback but before the new registry was similar, but `RuntimeError` was used instead

Reply:
Nope, there was a failure in the logic. Thanks for catching this